### ISSUE TITLE
3.0.3-enterprise doesn't start up

### ISSUE THREAD

**Issue by rattrap** [couchbase/docker#69]
Seems some golang binaries don't work on OS X and the server is not starting up:

```
docker run -d --name db -p 8091-8094:8091-8094 -p 11210:11210 couchbase:enterprise-3.0.3

root@59f520c81ba4:/opt/couchbase/bin# ./generate_cert
Trace/breakpoint trap
root@59f520c81ba4:/opt/couchbase/bin# ./gozip
Trace/breakpoint trap
root@59f520c81ba4:/opt/couchbase/bin# uname -a
Linux 59f520c81ba4 4.9.31-moby #1 SMP Sat Jun 24 06:29:23 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
MathBook:couchbase-cluster leo$ uname -a
Darwin MathBook 16.6.0 Darwin Kernel Version 16.6.0: Fri Apr 14 16:21:16 PDT 2017; root:xnu-3789.60.24~6/RELEASE_X86_64 x86_64
MathBook:couchbase-cluster leo$ docker -v
Docker version 17.06.0-ce, build 02c1d87
```

Having the same issue on Amazon Linux 7.

**Comment by rattrap** [comment]
Related to this - https://github.com/docker/for-mac/issues/1841

**Comment by rattrap** [comment]
Hello,

No. Without Docker. I just installed the binary rpm on AL.
We had the binary in our private RHEL repo since way back when :). Tried to download it from the official repo but it seems it's not available anymore.

**Comment by ceejatec** [comment]
I'd be curious to know if you see the same problem with the 3.1.6 download, which is still downloadable from couchbase.com. But as far as I can recall, those golang binaries were pre-compiled back in those days and quite possibly weren't actually changed between 3.0.3 and 3.1.6. And the version of golang used probably wasn't either, which in my experience is the main determinant of whether a binary will work or crash on a given platform. So there's a fair chance it'll still fail with 3.1.6. I can't test it myself as I don't have access to any machine exhibiting this problem, but I'll ask a coworker with a Mac to try it out.

In any case, I don't suspect this issue will get worked on. 3.0.3 is quite old (we have 5.0.0 beta 2 out now) and neither docker-on-mac nor Amazon Linux 7 are supported platforms for it. Also this clearly isn't a Docker-related issue so I feel I should probably close this issue on the couchbase/docker repository.

**Comment by rattrap** [comment]
Same issue with 3.1.6 and enterprise-3.1.6:

```
root@6f0737243261:/opt/couchbase/bin# ./generate_cert
Trace/breakpoint trap
root@6f0737243261:/opt/couchbase/bin# ./gozip
Trace/breakpoint trap
```

I know it's an old version but it's the one we use in production. Anyway, I eventually found RPM on the official repo and if I recompile the golang binaries it seems to work fine (https://github.com/rattrap/couchbase-cluster/blob/master/provisioning/couchbase.yml#L64). I just wanted to do some loadtests on different environments.

Thanks for your interest and we'll consider upgrading to a newer version although, when we tried this last year with couchbase 4.something, the nodes randomly crashed with OOM.
